data_structures/phantom_node.hpp

```diff
@@ -85,7 +88,11 @@ struct PhantomNode
     int forward_offset;
     int reverse_offset;
     unsigned packed_geometry_id;
-    unsigned component_id;
+    struct ComponentType {
+        uint32_t id : 31;
+        bool is_tiny : 1;
+    } component;
+    static_assert(sizeof(ComponentType) == 4, "ComponentType needs to 4 bytes big");
```

Comment:
Implementation defined size, since this is a struct with bit fields §9.6 Bit-fields N3936